### ISSUE TITLE
The sdk is not host agnostic anymore

### ISSUE THREAD

**Issue by ResolumeMenno** [resolume/ffgl#40]
There's quite a bit of code in the sdk that mentions Resolume specifically. FFGL is an open standard that should not care about what software it's being hosted in. If there's any workarounds or implementation that have to be made specifically for Resolume those implementations should be expanded so that this is no longer needed. Neither the sdk nor the sdk's examples should have any host specific code in them.

**Comment by spxl** [comment]
Do you mean workarounds in the SDK itself?

And what do you mean by "those implementations should be expanded" - that they should be catered for by the SDK, but in a way that is more generic to be used by other hosts (and mentions of Resolume removed so the SDK isn't "branded")?

**Comment by ResolumeMenno** [comment]
yes workarounds in the SDK. For example in FFGL2.1 i've already removed changing the texture wrapmodes to border clamping because supposedly 'there is a bug in resolume' as it's comment said. However this wasn't resolume related at all, it just surfaced because tests with the plugin were done in resolume. This specific issue is due to the FFGL standard where texture inputs are allowed to have a different width/height than their hardware width/height.

Some things i think should still be changed:
-FFGLPlugin.h mentions how Resolume shows parameters in the ui. If Resolume ever decides to change this the sdk's description will be incorrect, so it should be updated to not describe any host behaviour.
-FFGLPluginSDK.h has some 'additional ffgl functions by Resolume'. As a plugin developer it's not clear to me which functions these are, and what's happening in other hosts. Let's say it's the Connect/Disconnect functions. Can i put code in here if my plugin gets hosted in a different host? We should either make these functions part of FFGL itself (and thus require hosts to call them), or we should have some mechanism where it's clear to the plugin which part of the sdk is an extension, and if the currently running plugin instance supports the extension (eg is hosted inside a host supporting the extension)

As far as branding in the examples, i'm not too concerned. eg i think it's nice that in the CustomThumbnail example we mention that Resolume prefers a 160x120 thumbnail. It still mentions that the resolution may be different if the plugin so desires, but a default sensible for Resolume is fine here i think?
